apache/submarine
# SUBMARINE-180. E2E Test Framework

### What is this PR for?
To promise the functionality of workbench-web, we add the E2E test framework to this project.


### What type of PR is it?
Feature

### Todos
* Check the accuracy of E2E test framework on different OS.
* After the web page finishes, we will add more front-end test cases.

### What is the Jira issue?
* https://issues.apache.org/jira/browse/SUBMARINE-180

### How should this be tested?

* [CI Pass](https://travis-ci.org/apache/submarine/builds/604295591?utm_source=github_status&utm_medium=notification)

* Step1:
```
cd hadoop-submarine/submarine-test/e2e
```
* Step2-1: (Ubuntu)
```
mvn verify
``` 
* Step2-2: (MacOS)
```
mvn clean install -U
```
* After we execute the command in Step2, it will open the firefox browser, and then enter the string "admin" in the "id" and "password" blank.
* The result is showed in the following screenshot.
### Screenshots (if appropriate)
![螢幕快照 2019-10-29 上午12 57 48](https://user-images.githubusercontent.com/20109646/67699523-3d235c80-f9e7-11e9-9789-c6249e25159f.png)

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? Yes
